packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -577,22 +578,24 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
      *
      * @param widget widget file resource of which should be revealed and selected
      */
-    async selectWidgetFileNode(widget: Widget | undefined): Promise<void> {
+    async selectWidgetFileNode(widget: Widget | undefined, isSelectedByEditor?: boolean): Promise<void> {
         if (Navigatable.is(widget)) {
             const resourceUri = widget.getResourceUri();
             if (resourceUri) {
                 const { model } = await this.widget;
                 const node = await model.revealFile(resourceUri);
                 if (SelectableTreeNode.is(node)) {
-                    model.selectNode(node);
+                    model.selectNode(node, isSelectedByEditor ? true : false);
                 }
             }
         }
     }
 
-    protected onCurrentWidgetChangedHandler(): void {
+    protected onCurrentWidgetChangedHandler(args: FocusTracker.IChangedArgs<Widget>): void {
```

Comment:
This is a breaking change that should probably be documented (in the changelog).